eth/downloader/downloader.go

```diff
@@ -483,63 +415,10 @@ func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td *big.I
 	d.syncStatsChainHeight = height
 	d.syncStatsLock.Unlock()
 
-	// Ensure our origin point is below any fast sync pivot point
-	if mode == FastSync {
-		if height <= uint64(fsMinFullBlocks) {
-			origin = 0
-		} else {
-			pivotNumber := pivot.Number().Uint64()
-			if pivotNumber <= origin {
-				origin = pivotNumber - 1
-			}
-			// Write out the pivot into the database so a rollback beyond it will
-			// reenable fast sync
-			rawdb.WriteLastPivotNumber(d.stateDB, pivotNumber)
-		}
-	}
 	d.committed = 1
-	if mode == FastSync && pivot.Number().Uint64() != 0 {
+	if pivot.Number().Uint64() != 0 {
 		d.committed = 0
```

Comment:
I think this is wrong. If `mode == FastSync` is false, then we should do nothing

Reply:
Yes, Removed the logic.